src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -60,7 +60,20 @@ public static boolean isNonZeroUnsignedInteger(String s) {
 
         try {
             int value = Integer.parseInt(s);
-            return value > 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
+            // "+1" is successfully parsed by Integer#parseInt(String)
+            return value > 0 && !s.startsWith("+") && !s.contains("-");
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+    }
+
+    public static boolean isNonNegativeInteger(String s) {
```

Comment:
From the checkstyle error messages, I think maybe you should add a javadoc comment here?